### PR TITLE
[Feature] Kaiko WS Handler

### PR DESCRIPTION
**Pull request by justinkaseman**
### Description
WS handler for Kaiko

### Draft Status
Seeing 403 errors when trying to subscribe while testing with `staging-kaiko-api-key`.

Kaiko support has been sent an email.

### Story
https://app.clubhouse.io/chainlinklabs/story/9254/websocket-handler-kaiko

### CODE REVIEW

File: packages/core/bootstrap/src/lib/ws/index.ts [hunk 1]
```diff
@@ -18,9 +18,11 @@ export const withWebSockets = (
 
   const wsHandler = await makeWsHandler()
   const wsConfig = getWSConfig()
+  console.log(wsConfig, wsHandler)
```
Comment (krebernisak) [inline packages/core/bootstrap/src/lib/ws/index.ts:21]: This seems as debug log messages, and this should already be logged by WS epics.
Consider removing this extra logging.

**Review by krebernisak — COMMENTED** [review]
Looks good! 

@justinkaseman what's the status of this?
If we can quickly merge, we could use it as soon as today for the next phase of the Optimism WS test.

File: packages/core/bootstrap/README.md [hunk 2]
```diff
@@ -210,7 +210,16 @@ From the moment the subscription is confirmed, the adapter will start receiving
 
 | Required? |                Name                |                                                                             Description                                                                              | Options | Defaults to |
 | :-------: | :--------------------------------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------: | :-----: | :---------: |
+|           |        `WS_CONNECTION_KEY`         |                                                                                                                                                                      |         |  undefined  |
```
Comment (RodrigoAD) [inline packages/core/bootstrap/README.md:213]: I don't know if you want to include this env vars, they don't have any effect yet

File: packages/sources/kaiko/src/adapter.ts [hunk 3]
```diff
@@ -75,3 +82,71 @@ export const execute: ExecuteWithConfig<Config> = async (request, config) => {
...
+    if (validator.error) return
+    const base = validator.validated.data.base.toUpperCase()
+    const quote = validator.validated.data.quote.toUpperCase()
+    return `${base}-${quote}`
+  }
+  return () => {
+    const defaultConfig = config || makeConfig()
+    return {
+      connection: {
+        url: defaultConfig.api.baseWsURL || DEFAULT_WS_API_ENDPOINT,
```
Comment (RodrigoAD) [inline packages/sources/kaiko/src/adapter.ts:138]: Is the authentication missing?

Comment (justinkaseman) [reply packages/sources/kaiko/src/adapter.ts:138]: Must be, I was having problems connecting.

File: packages/core/bootstrap/src/lib/ws/index.ts (hunk 1 above)
Comment (justinkaseman) [reply packages/core/bootstrap/src/lib/ws/index.ts:21]: It _is_ extra logging that shouldn't have been committed, removing.

File: packages/sources/kaiko/README.md [hunk 4]
```diff
@@ -2,9 +2,10 @@
 
 ### Environment Variables
 
-| Required? |  Name   |                                 Description                                 | Options | Defaults to |
-| :-------: | :-----: | :-------------------------------------------------------------------------: | :-----: | :---------: |
-|    ✅     | API_KEY | An API key that can be obtained from [here](https://www.coinapi.io/pricing) |         |             |
+| Required? |       Name        |                     Description                      | Options | Defaults to |
+| :-------: | :---------------: | :--------------------------------------------------: | :-----: | :---------: |
+|    ✅     |      API_KEY      |      An API key that can be obtained from Kaiko      |         |             |
+|    ✅     | WS_CONNECTION_KEY | An WS connection key that can be obtained from Kaiko |         |             |
```
Comment (RodrigoAD) [inline packages/sources/kaiko/README.md:8]: `WS_CONNECTION_KEY` is not necessary. The `API_KEY` is the same for WS and HTTP

Comment (justinkaseman) [reply packages/sources/kaiko/README.md:8]: Forgot to take this one out, thanks for catching that.

File: packages/core/bootstrap/README.md [hunk 5]
```diff
@@ -216,7 +216,14 @@ From the moment the subscription is confirmed, the adapter will start receiving
...
 | :-------: | :--------------------------------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------: | :-----: | :---------: |
 |           |            `WS_ENABLED`            |                                               Set this to `true` to enable WS support (on adapters that support this).                                               |         |   `false`   |
+|           |        `WS_CONNECTION_KEY`         |                                                                                                                                                                      |         |  undefined  |
+|           |       `WS_CONNECTION_LIMIT`        |                                                                                                                                                                      |         |     `1`     |
```
Comment (HenryNguyen5) [inline packages/core/bootstrap/README.md:220]: We should add these to packages/core/bootstrap/schemas/env.json

**Pull request closed without merging**